tests/course/test_for_courses.py

```diff
@@ -0,0 +1,181 @@
+
+
+# student 2 / TEACHER = USER0 , STUDENT = USER1
+@pytest.fixture
+def studentCourses1(db, teacherCourses1, user1):
+    student_course = StudentCourses(student_id=user1, teacher_course_id=teacherCourses1)
+    student_course.save()
+    return student_course
+
+
+# student 3 / TEACHER = USER1 , STUDENT = USER0
+@pytest.fixture
+def studentCourses2(db, teacherCourses2, user0):
+    user0.save()
+    student_course = StudentCourses(student_id=user0, teacher_course_id=teacherCourses2)
+    student_course.save()
+    return student_course
+
+
+# VALID
+PENDING = 'Pending'
+CONFIRMED = 'Confirmed'
+COURSE_NAME = "math course"
+DESCRIPTION = "this is a math course"
+DIFFICULTY = 'A'
+CATEGORY = "MATHS"
+PRICE = 100
+YEARS_OF_EXP = 3
+INVALID_STATUS = 'not status'
+
+
+class TestStudentCourse:
+
+    # CHECK SAVE AND DELETE
+
+    def test_new_student_course(self, studentCourses0, user1, teacherCourses1):
+        assert studentCourses0.status == PENDING
+        assert studentCourses0.student_id == user1
+        assert studentCourses0.teacher_course_id == teacherCourses1
+
+    @pytest.mark.django_db()
+    def test_persist_student_course(self, studentCourses0):
+        studentCourses0.save()
+        assert studentCourses0 in StudentCourses.objects.all()
+
+    @pytest.mark.django_db()
+    def test_delete_ref_member0(self, studentCourses0):
+        self.test_persist_student_course(studentCourses0)
+        studentCourses0.teacher_course_id.delete()
+        assert studentCourses0 not in StudentCourses.objects.all()
+
+    @pytest.mark.django_db()
+    def test_delete_ref_member1(self, studentCourses0):
+        self.test_persist_student_course(studentCourses0)
+        studentCourses0.student_id.delete()
+        assert studentCourses0 not in StudentCourses.objects.all()
+
+    # CHECK FILTERS AND METHODS
+
+    def test_change_to_confirmed(self, studentCourses1):
+        assert studentCourses1.status == PENDING
+        studentCourses1.change_to_confirmed()
+        assert studentCourses1.status == CONFIRMED
+
+    def test_teacher_pending(self, studentCourses1, studentCourses2, user0, user1):
+        # USER0 AS TEACHER
+        assert studentCourses1.teacher_course_id.course_id in StudentCourses.objects.get_teacher_pending(user0.pk)
+        assert studentCourses2.teacher_course_id.course_id not in StudentCourses.objects.get_teacher_pending(user0.pk)
+        studentCourses1.change_to_confirmed()
+        assert studentCourses1.teacher_course_id.course_id not in StudentCourses.objects.get_teacher_pending(user0.pk)
+        # USER1 AS TEACHER
+        assert studentCourses2.teacher_course_id.course_id in StudentCourses.objects.get_teacher_pending(user1.pk)
+        assert studentCourses1.teacher_course_id.course_id not in StudentCourses.objects.get_teacher_pending(user1.pk)
+        studentCourses2.change_to_confirmed()
+        assert studentCourses1.teacher_course_id.course_id not in StudentCourses.objects.get_teacher_pending(user1.pk)
```

Comment:
I would suggest using `pytest.mark.parametrize` (read about it), but since you're using fixtures it'll be less suitable. the test is ok, but maybe I'll divide it to smaller tests (e.g: one for `not in` and one for `in`), or at least test with lists (e.g: [x.y.z] no in). check if you can improve it, but if not that's fine.